CDT/CMakeLists.txt

```
@@ -155,8 +148,12 @@ target_compile_definitions(
 )
 
 if(CDT_USE_BOOST)
-    target_link_libraries(${PROJECT_NAME} INTERFACE Boost::boost)
-endif()
+    if(CDT_USE_AS_COMPILED_LIBRARY)
+        target_link_libraries(${PROJECT_NAME} PUBLIC Boost::boost)
+    else ()
+        target_link_libraries(${PROJECT_NAME} INTERFACE Boost::boost)
+    endif ()
```

```suggestion
    target_link_libraries(${PROJECT_NAME} ${cdt_scope} Boost::boost)
```
Like everywhere else